### ISSUE TITLE
Build errors on latest rust nightly

### ISSUE THREAD

**Issue by cevn** [gchp/rustbox#18]
```
   Compiling rustbox v0.2.9
/Users/sameerdhar/.cargo/registry/src/github.com-1ecc6299db9ec823/rustbox-0.2.9/src/rustbox.rs:122:32: 122:37 error: explicit lifetime bound required
/Users/sameerdhar/.cargo/registry/src/github.com-1ecc6299db9ec823/rustbox-0.2.9/src/rustbox.rs:122     Opt(InitOption, Option<Box<Error>>),
                                                                                                                                  ^~~~~
error: aborting due to previous error
Could not compile `rustbox`.
```

```
$[~] rustc --version                            
rustc 1.0.0-nightly (29bd9a06e 2015-01-20 23:03:09 +0000)
```

**Comment by gchp** [comment]
Yep, I'm seeing this too. Looking into it.

**Comment by gchp** [comment]
I really dislike debugging rust's lifetime errors :sweat_smile:

**Comment by cevn** [comment]
You and me both, I tried myself for a bit and had literally no idea what was going on, but I am still learning. Maybe go ask in #rust on irc.mozilla.org?

**Comment by pythonesque** [comment]
@gchp: I think `Error` needs either `'static` or `Send`.

**Comment by cevn** [comment]
replacing line 122 with this: 
```
    Opt(InitOption, Option<Box<Error + 'static>>),
```

produces this output:

```
   Compiling rustbox v0.2.9 (file:///Users/sameerdhar/Documents/projects/personal/rust/rustbox)
src/rustbox.rs:291:29: 291:31 error: cannot infer an appropriate lifetime for automatic coercion due to conflicting requirements
src/rustbox.rs:291                             rg)
                                               ^~
<std macros>:1:1: 6:60 note: in expansion of try!
src/rustbox.rs:287:32: 292:85 note: expansion site
src/rustbox.rs:291:29: 291:31 note: first, the lifetime cannot outlive the expression at 291:28...
src/rustbox.rs:291                             rg)
                                               ^~
<std macros>:1:1: 6:60 note: in expansion of try!
src/rustbox.rs:287:32: 292:85 note: expansion site
src/rustbox.rs:291:29: 291:31 note: ...so that auto-reference is valid at the time of borrow
src/rustbox.rs:291                             rg)
                                               ^~
<std macros>:1:1: 6:60 note: in expansion of try!
src/rustbox.rs:287:32: 292:85 note: expansion site
src/rustbox.rs:287:37: 291:32 note: but, the lifetime must be valid for the call at 287:36...
src/rustbox.rs:287                 *stderr = Some(try!(redirect(
src/rustbox.rs:288                             try!(PipeStream::open(libc::STDERR_FILENO)
src/rustbox.rs:289                             .map_err( |e| InitError::Opt(InitOption::BufferStderr,
src/rustbox.rs:290                                                          Some(Box::new(e) as Box<Error + 'static>)) )),
src/rustbox.rs:291                             rg)
<std macros>:1:1: 6:60 note: in expansion of try!
src/rustbox.rs:287:32: 292:85 note: expansion site
src/rustbox.rs:291:29: 291:31 note: ...so that argument is valid for the call
src/rustbox.rs:291                             rg)
                                               ^~
<std macros>:1:1: 6:60 note: in expansion of try!
src/rustbox.rs:287:32: 292:85 note: expansion site
error: aborting due to previous error
Could not compile `rustbox`.
```

What do you mean by using Send?

**Comment by bharadwaj6** [comment]
@cevn: I am also facing this issue. He's talking about [Send](http://doc.rust-lang.org/0.12.0/core/kinds/trait.Send.html) kind. Read [this explanation](http://smallcultfollowing.com/babysteps/blog/2014/02/28/rust-rfc-opt-in-builtin-traits/), and [this](http://stackoverflow.com/q/25959075/1076075) is how it's normally used.

I am not sure how to fix this error though... I guess we have to read the docs and dig deeper to get it done.

**Comment by pythonesque** [comment]
`redirect` in that file also has `Box<Error>` in its signature.  It will probably require changing in a number of places (you may have tried this already).

**Comment by cevn** [comment]
Oh good call @pythonesque, that was enough to fix it. I'll make a PR

**Comment by gchp** [comment]
Just released `0.2.10` with the fix for this.
